### ISSUE TITLE
Wrong configurations kill auth daemon without notifying the service

### ISSUE THREAD

**Issue by BraulioV** [wazuh/wazuh#4202]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 3.10.2-31014 | Authd/ossec-control/service | Manager | Packages/Sources | CentOS 7 |

Hi team, 

While testing the different options available with the auth daemon I found an error related to its configuration, the `ossec-control` and the service. 

Using this configuration to enable the agents' verification:
```xml
  <auth>
    <disabled>no</disabled>
    <port>1515</port>
    <use_source_ip>yes</use_source_ip>
    <force_insert>yes</force_insert>
    <force_time>0</force_time>
    <purge>yes</purge>
    <use_password>no</use_password>
    <limit_maxagents>yes</limit_maxagents>
    <ciphers>HIGH:!ADH:!EXP:!MD5:!RC4:!3DES:!CAMELLIA:@STRENGTH</ciphers>
    <ssl_agent_ca>/var/ossec/etc/etc/rootCA.pem</ssl_agent_ca>
    <ssl_verify_host>no</ssl_verify_host>
    <ssl_manager_cert>/var/ossec/etc/sslmanager.cert</ssl_manager_cert>
    <ssl_manager_key>/var/ossec/etc/sslmanager.key</ssl_manager_key>
    <ssl_auto_negotiate>no</ssl_auto_negotiate>
  </auth>
```
and storing the `rootCA.pem` file in `/var/ossec/etc/rootCA.pem` cause the following error:
```
2019/11/07 16:59:13 ossec-authd: ERROR: Unable to read CA certificate file "/var/ossec/etc/etc/rootCA.pem"
2019/11/07 16:59:13 ossec-authd: ERROR: SSL error. Exiting.
```
and `ossec-authd` stops its execution.

This works as expected, but, what is not expected is to restart the manager using `ossec-control` or the service manager and get the service as up and running. 
![image](https://user-images.githubusercontent.com/9081114/68411855-8f087700-018b-11ea-8b4b-a8672a7fe306.png)

The `start` function from `ossec-control` performs a syntax check at the beginning but not a semantic check. The syntax is correct but its semantic, not. Later, when `ossec-authd` is started in https://github.com/wazuh/wazuh/blob/7aa4d46661f8b0f97b0969634482626f30866731/src/init/ossec-server.sh#L337-L371 the script checks the return of running `/var/ossec/bin/ossec-authd`, which is 0 and then, pass to the next daemon. A bit later, the daemon dies after trying to open the rootCA file, but, for the `ossec-control` and the service, everything worked as expected.

In summary, these are the problems:
- `ossec-authd` doesn't check if all the files from its configuration are available until it is too late.
- `ossec-control` doesn't correctly check the execution of the daemon.

**Comment by chemamartinez** [comment]
Solved at #4237
